src/api.py

```
@@ -0,0 +1,154 @@
+        
+        logger.info("Request processed successfully",
+                   request_id=request.request_id,
+                   processing_time_ms=response.processing_time_ms)
+        
+        return response
+        
+    except Exception as e:
+        logger.error("Processing failed",
+                    request_id=request.request_id,
+                    error=str(e))
+        raise HTTPException(status_code=500, detail=str(e))
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)